tests/test_shell.py

```diff
@@ -235,6 +235,19 @@ def test_view(self):
                 "view/a/{}/job".format(sp["a"])
             ) == os.path.realpath(project.open_job(sp).workspace())
 
+    @pytest.mark.skipif(WINDOWS, reason="Symbolic links are unsupported on Windows.")
+    def test_view_incomplete_path_spec(self):
+        self.call("python -m signac init my_project".split())
+        project = signac.Project()
+        sps = [{"a": i} for i in range(3)]
+        for sp in sps:
+            project.open_job(sp).init()
+        os.mkdir("view")
+
+        # Should error if user-provided path doesn't make 1-1 mapping
+        err = self.call("python -m signac view non_unique".split(), error=True)
```

Comment:
This is the code that is not working as expected

Reply:
The `error` parameters make it so that the STDERR instead of the STDOUT stream is returned. You probably meant to use `raise_error`, see line 68.

Reply:
I left a breakpoint in the call function and `out` and `err` from `p.communicate()` are both empty strings and `p.returncode` is 0. Does that mean the process doesn't think there is an error? When I run the command on the same terminal I'm testing from I get the error as expected.

Reply:
```
$ python -m signac view --help
usage: __main__.py view [-h] [-f FILTER [FILTER ...]]
                        [-d DOC_FILTER [DOC_FILTER ...]]
                        [-j JOB_ID [JOB_ID ...]] [-i INDEX]
                        [prefix] [path]

Generate a human readable set of paths representing state points in the
workspace, e.g.
view/param_name_1/param_value_1/param_name_2/param_value_2/job. The leaf nodes
of this directory structure are symlinks (named "job") into the workspace
directory for that parameter combination. Note that all positional arguments
must be provided before any keyword arguments. In particular, the prefix and
path must be specified before arguments such as the filters, e.g. signac view
$PREFIX $VIEW_PATH -f FILTERS -d DOC_FILTERS.

positional arguments:
  prefix                The path where the view is to be created. Defaults to
                        view.
  path                  The path used for the generation of the linked view
                        hierarchy, defaults to '{{auto}}' (see
                        Project.export_to for information on how this is
                        expanded).

optional arguments:
  -h, --help            show this help message and exit

select:
  -f FILTER [FILTER ...], --filter FILTER [FILTER ...]
                        Limit the view to jobs matching this state point
                        filter.
  -d DOC_FILTER [DOC_FILTER ...], --doc-filter DOC_FILTER [DOC_FILTER ...]
                        Limit the view to jobs matching this document filter.
  -j JOB_ID [JOB_ID ...], --job-id JOB_ID [JOB_ID ...]
                        Limit the view to jobs with these job ids.
  -i INDEX, --index INDEX
                        The filename of an index file.
```

This test is using `non_unique` as the _prefix_, not the _path_. You need to give `view` as the prefix and `non_unique` as the path:
```suggestion
        err = self.call("python -m signac view view non_unique".split(), error=True)
```

Reply:
@csadorf is also correct. My suggestion needs the `raise_error` flag:
```suggestion
        err = self.call(
            "python -m signac view view non_unique".split(),
            error=True,
            raise_error=False,
        )
```

I'm going to commit this change so we can get CI passing.

Reply:
Argh, I knew it was something obvious! Thank you

Reply:
Not obvious! I think this is an annoying CLI design choice to require the prefix before specifying the path. I always make this mistake and forget to write `signac view view path` (because that sounds silly). Could be fixed in signac 2.0 if we want. Filing an issue.

Reply:
I filed #653 for this. I think it's a flaw in the CLI design.

Reply:
> I think this is an annoying CLI design choice to require the prefix before specifying the path.

Sorry. 😁